trieste/bayesian_optimizer.py

```diff
@@ -96,18 +98,26 @@ class Record(Generic[StateType]):
     @property
     def dataset(self) -> Dataset:
         """The dataset when there is just one dataset."""
-        if len(self.datasets) == 1:
-            return next(iter(self.datasets.values()))
+        # Ignore local datasets.
+        datasets: Mapping[Tag, Dataset] = dict(
```

Comment:
More Pythonic to use comprehension?

```suggestion
        datasets: Mapping[Tag, Dataset] = {
            k : v for k, v in self.datasets.items()
            if not LocalizedTag.from_tag(k).is_local
        }
```

Reply:
Update: since you use this repeatedly, it might even be worth defining a helper function in acquisition.utils

```python
def ignoring_local_models(model_map: Mapping[Tag, T]) -> Mapping[Tag, T]:
    return { ... }
```